frontend/src/app/office-hours/office-hours.service.ts

```diff
@@ -0,0 +1,367 @@
+/**
+ * The Office Hours Service abstracts HTTP requests to the backend
+ * from the components.
+ *
+ * @author Sadie Amato, Madelyn Andrews, Bailey DeSouza, Meghan Sun
+ * @copyright 2024
+ * @license MIT
+ */
+
+import { Injectable } from '@angular/core';
+import { HttpClient, HttpHeaders } from '@angular/common/http';
+import { AuthenticationService } from '../authentication.service';
+import { MatSnackBar } from '@angular/material/snack-bar';
+import { Observable, of } from 'rxjs';
+import {
+  OfficeHoursSectionDetails,
+  OfficeHoursSectionPartial,
+  OfficeHoursSectionDraft,
+  OfficeHoursEventDetails,
+  OfficeHoursEventDraft,
+  TicketDetails,
+  TicketDraft,
+  OfficeHoursEvent,
+  OfficeHoursEventType,
+  TicketType,
+  TicketState,
+  TicketPartial,
+  Ticket,
+  OfficeHoursEventStatus,
+  StudentOfficeHoursEventStatus,
+  OfficeHoursSection,
+  OfficeHoursSectionTrailingData,
+  OfficeHoursEventModeType,
+  Weekday,
+  OfficeHoursEventPartial
+} from './office-hours.models';
+import {
+  Section,
+  SectionMember,
+  SectionMemberPartial,
+  Term
+} from '../academics/academics.models';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class OfficeHoursService {
+  constructor(
+    protected http: HttpClient,
+    protected auth: AuthenticationService,
+    protected snackBar: MatSnackBar
+  ) {}
+
+  /* Ticket-Related Methods: */
+
+  createTicket(ticket_draft: TicketDraft): Observable<TicketDetails> {
```

Comment:
Nit: Would recommend switching to camelCase where possible (in this file, for the parameters)